numpy/_core/tests/examples/cython/meson.build

```diff
@@ -34,3 +35,20 @@ py.extension_module(
     ],
     include_directories: [npy_include_path],
 )
+
+# py does not export the is_pypy method
+if not py.get_variable('SOABI').startswith('pypy')
```

Comment:
There must be a better way to figure out if we are building on PyPy. The problem this addresses is that PyPy does not ship a limited API link library `python3.lib` on windows.

Reply:
I don't think that there is right now, running this in a subprocess call is maybe slightly more future-proof but also much more verbose and slower, so not better:
```bash
python -c "import sys; print(sys.implementation.name)"
```

It shouldn't be hard to extend https://mesonbuild.com/Python-module.html#methods with an `implementation_name()` method I think. Should be quite rare that it's needed, but it does make sense.

Reply:
I opened an issue on meson https://github.com/mesonbuild/meson/issues/12727

Reply:
I refactored this PR to avoid the problem: I moved the test to the limited_api tests and skipped that test if run under PyPy.